storage/remote/queue_manager.go

```diff
@@ -1156,10 +1181,25 @@ func (s *shards) sendSamplesWithBackoff(ctx context.Context, samples []prompb.Ti
 	reqSize := len(*buf)
 	*buf = req
 
-	// An anonymous function allows us to defer the completion of our per-try spans
-	// without causing a memory leak, and it has the nice effect of not propagating any
-	// parameters for sendSamplesWithBackoff/3.
-	attemptStore := func(try int) error {
+	var (
+		bufWithoutTs      *[]byte
+		containsExemplars bool
+	)
+	if s.qm.cfg.RetryPolicy != nil {
+		exemplars := removeTs(samples)
+		if len(exemplars) > 0 {
```

Comment:
you could also just check the input parameter `exemplarCount int`

Reply:
Isn't it safer to consider the actual length received than function param? If you think we should use the param, let me know and I will update.